cmd/kubectl-gadget/advise/network-policy.go

```diff
@@ -94,11 +95,16 @@ func runNetworkPolicyMonitor(cmd *cobra.Command, args []string) error {
 		TraceOutputState: gadgetv1alpha1.TraceStateStarted,
 		CommonFlags:      &params,
 	}
+
+	var mu sync.Mutex
+
 	count := 0
 	transform := func(line string) string {
 		line = strings.Replace(line, "\r", "\n", -1)
+		mu.Lock()
 		w.Write([]byte(line))
 		w.Flush()
```

Comment:
nit: Can the `.Flush()` be moved after the `.Unlock()`?

Reply:
I think the operation of writing and flushing should be "atomic" - it seems intended that a flush should happen after each line (and it's also important because the underlying buffer is limited) - which would not be guaranteed when calling it outside the lock.